mypy/checker.py

```diff
@@ -2471,7 +2499,22 @@ def find_isinstance_check(node: Node,
         _, if_vars = conditional_type_map(node, vartype, NoneTyp(), weak=weak)
         return if_vars, {}
     elif isinstance(node, OpExpr) and node.op == 'and':
-        left_if_vars, right_else_vars = find_isinstance_check(
+        left_if_vars, left_else_vars = find_isinstance_check(
+            node.left,
+            type_map,
+            weak,
+        )
+
+        right_if_vars, right_else_vars = find_isinstance_check(
+            node.right,
+            type_map,
+            weak,
+        )
+
+        return (and_conditional_maps(left_if_vars, right_if_vars),
```

Comment:
Could you try to describe this as it's a little tricky, or maybe give an example (doesn't need to cover all special cases, but just the general idea).